flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapKeyedStateBackend.java

```diff
@@ -333,13 +334,17 @@ public SavepointResources<K> savepoint() {
     }
 
     @Override
-    public void notifyCheckpointComplete(long checkpointId) {
-        // Nothing to do
+    public void notifyCheckpointComplete(long checkpointId) throws Exception {
+        if (checkpointStrategy instanceof CheckpointListener) {
+            ((CheckpointListener) checkpointStrategy).notifyCheckpointComplete(checkpointId);
+        }
     }
 
     @Override
-    public void notifyCheckpointAborted(long checkpointId) {
-        // nothing to do
+    public void notifyCheckpointAborted(long checkpointId) throws Exception {
```

Comment:
`notifyCheckpointAborted()` is not implemented/used in `IncrementalHeapSnapshotStrategy` , is it necessary to modify this method?

Reply:
You're right, it's not implemented by the strategy. But I think it would be more consistent to have it implemented here the same way as `notifyCheckpointComplete`. 

WDYT?